java/client/src/main/java/glide/api/models/commands/RangeOptions.java

```diff
@@ -0,0 +1,257 @@
+
+        public String toArgs() {
+            return redisApi;
+        }
+    }
+
+    /** Represents a specific numeric score boundary in a sorted set. */
+    public static class ScoreBoundary implements ScoreRange {
+        /** The score value. */
+        private final double bound;
+
+        /** Whether the score value is inclusive. Defaults to true if not set. */
+        private final boolean isInclusive;
+
+        public ScoreBoundary(double bound, boolean isInclusive) {
+            this.bound = bound;
+            this.isInclusive = isInclusive;
+        }
+
+        public ScoreBoundary(double bound) {
+            this.bound = bound;
+            this.isInclusive = true;
+        }
+
+        /** Convert the score boundary to the Redis protocol format. */
+        public String toArgs() {
+            return this.isInclusive ? String.valueOf(this.bound) : "(" + this.bound;
+        }
+    }
+
+    public interface LexRange {
+        String toArgs();
+    }
+
+    /**
+     * Enumeration representing lexicographic positive and negative infinity bounds for sorted set.
+     */
+    @RequiredArgsConstructor
+    public enum InfLexBound implements LexRange {
+        POSITIVE_INFINITY("+"),
+        NEGATIVE_INFINITY("-");
+
+        private final String redisApi;
+
+        @Override
+        public String toArgs() {
+            return redisApi;
+        }
+    }
+
+    /** Represents a specific lexicographic boundary in a sorted set. */
+    public static class LexBoundary implements LexRange {
+        /** The lex value. */
+        private final String value;
+
+        /** Whether the lex value is inclusive. Defaults to true if not set. */
+        private final boolean isInclusive;
+
+        public LexBoundary(@NonNull String value, boolean isInclusive) {
+            this.value = value;
+            this.isInclusive = isInclusive;
+        }
+
+        public LexBoundary(@NonNull String value) {
+            this.value = value;
+            this.isInclusive = true;
+        }
+
+        /** Convert the lex boundary to the Redis protocol format. */
+        @Override
+        public String toArgs() {
+            return this.isInclusive ? "[" + this.value : "(" + this.value;
+        }
+    }
+
+    /**
+     * Represents a limit argument for a range query in a sorted set to be used in <a
+     * href="https://redis.io/commands/zrange">ZRANGE</a> command.<br>
+     * The optional LIMIT argument can be used to obtain a sub-range from the matching elements
+     * (similar to SELECT LIMIT offset, count in SQL).
+     */
+    @RequiredArgsConstructor
+    @Getter
+    public static class Limit {
+        /** The offset from the start of the range. */
+        private final long offset;
+
+        /**
+         * The number of elements to include in the range. A negative count returns all elements from
+         * the offset.
+         */
+        private final long count;
+    }
+
+    public interface RangeQuery {
```

Comment:
javadoc

Reply:
Don't know what to document for these interfaces they were created to be able to mimic the same functionality of Python and Node. Any ideas would help.

Reply:
```
    /**
     * Basic interface. Please use one of the following implementations:
     *
     * <ul>
     *   <li>{@link RangeByIndex}
     *   <li>{@link RangeByScore}
     *   <li>{@link RangeByLex}
     * </ul>
     */
```
Copied from routing.

Reply:
Thanks appreciate the guidance.